R/commonOrdinalRestrictions.R

```diff
@@ -0,0 +1,1190 @@
+  container[["coefficientsTable"]] <- coefficientsTable
+
+  coefs <- model[["coefficients"]]
+
+  if(!isTryError(coefs)) {
+
+    coefficientsTable$addColumnInfo(name = "coef",     title = gettext("Coefficient"), type = "string")
+    coefficientsTable$addColumnInfo(name = "estimate", title = gettext("Estimate"),    type = "number")
+    if(!is.null(coefs[["se"]]))
+      coefficientsTable$addColumnInfo(name = "se",     title = gettext("SE"),          type = "number")
+    if(!is.null(coefs[["t"]]))
+      coefficientsTable$addColumnInfo(name = "t",      title = gettext("t"),           type = "number")
+    if(!is.null(coefs[["p"]]))
+      coefficientsTable$addColumnInfo(name = "p",      title = gettext("p"),           type = "pvalue")
+
+
+    if(!is.null(model[["bootstrapSamples"]])) {
+      overtitle <- gettextf("%s%% CI", 100*options[["restrictedBootstrappingConfidenceIntervalLevel"]])
+      coefficientsTable$addColumnInfo(name = "lower",    title = gettext("Lower"),       type = "number", overtitle = overtitle)
+      coefficientsTable$addColumnInfo(name = "upper",    title = gettext("Upper"),       type = "number", overtitle = overtitle)
+      coefficientsTable$addFootnote(gettextf("Estimates based on %s successful bootstrap replicates.", model[["bootstrapSamples"]]))
+    }
+
+    coefficientsTable$setData(coefs)
+  } else {
+    message <- .aorExtractErrorMessageSoft(coefs)
+    coefficientsTable$setError(gettextf("Could not compute table of coefficients. Error message: %s", message))
+  }
+}
+
+## Informative Hypothesis tests ----
+.aorInformativeHypothesisTest <- function(container, options, model) {
+  if(!is.null(container[["ihtTable"]]) || !model[["informedHypothesisTest"]]) return()
+
+  ihtTable <- createJaspTable(title = gettext("Informative Hypothesis Tests"), position = 3)
+  container[["ihtTable"]] <- ihtTable
+  ihtTable$showSpecifiedColumnsOnly <- TRUE
+
+  ihtTable$addColumnInfo(name = "typeName", title = gettext("Hypothesis"),      type = "string", combine = TRUE)
+  ihtTable$addColumnInfo(name = "test",     title = gettext("Test"),            type = "string", combine = TRUE)
+  ihtTable$addColumnInfo(name = "stat",     title = gettext("Test Statistic"),  type = "number")
+  ihtTable$addColumnInfo(name = "pvalue",   title = gettext("p"),               type = "pvalue")
+
+  result <- try(.aorCalculateIHT(model[["fit"]]))
+
+  if(!isTryError(result)) {
+    result[["typeName"]] <- gettextf("Type %s", result[["type"]])
+    ihtTable$setData(result)
+    ihtTable$addCitation(c(
+      "Silvapulle, M. J. & Sen, P. K. (2005). Constrained statistical inference: Order, inequality, and shape constraints. Hoboken, NJ: Wiley.",
+      "Vanbrabant, L. & Rosseel, Y. (2020). Restricted statistical estimation and inference for linear models. http://restriktor.org"
+    ))
+    # for some reason, this takes forever
+    # .aorAddInformativeHypothesisTestFootnotes(ihtTable, result)
```

Comment:
I have a problem with this function. It adds a bunch of footnotes to the table. If I run it in jaspTools, it works fine, but if run it in JASP, the analysis runs forever and seemingly never gets done. So that seems like a some issue in the js?